internal/resources/policy/kind/network/input_flatten_test.go

```diff
@@ -0,0 +1,184 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package policykindnetwork
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/helper"
+	policyrecipenetworkmodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/policy/recipe/network"
+	policyrecipenetworkcommonmodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/policy/recipe/network/common"
+	reciperesource "github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/policy/kind/network/recipe"
+)
+
+func TestFlattenInput(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		description string
+		input       *inputRecipe
+		expected    []interface{}
+	}{
+		{
+			description: "check for nil input",
+			input:       nil,
+			expected:    nil,
+		},
+		{
+			description: "normal scenario with input as allow-all recipe",
+			input: &inputRecipe{
+				recipe: AllowAllRecipe,
+				inputAllowAll: &policyrecipenetworkmodel.V1alpha1CommonPolicySpecNetworkV1AllowAll{
+					FromOwnNamespace: helper.BoolPointer(true),
+				},
+			},
+			expected: []interface{}{
+				map[string]interface{}{
+					reciperesource.AllowAllKey: []interface{}{
+						map[string]interface{}{
+							reciperesource.FromOwnNamespaceKey: true,
+						},
+					},
+				},
+			},
+		},
+		{
+			description: "normal scenario with input as allow-all-to-pods recipe",
+			input: &inputRecipe{
+				recipe: AllowAllToPodsRecipe,
+				inputAllowAllToPods: &policyrecipenetworkmodel.V1alpha1CommonPolicySpecNetworkV1AllowAllToPods{
+					FromOwnNamespace: helper.BoolPointer(true),
+					ToPodLabels: []*policyrecipenetworkcommonmodel.V1alpha1CommonPolicySpecNetworkV1Labels{
+						{
+							Key:   "foo",
```

Comment:
the change from key , value to map based key:value , would u be taking it separately ?

Reply:
That is done already, please check the schema. The test will take value as the models are defined in a different way, but the expected result verifies that it has been changed to map bases key:value.